ogr/ogrsf_frmts/hana/ogrhanalayer.cpp

```diff
@@ -114,65 +114,75 @@ CreateFieldDefn(const AttributeColumnDescription &columnDesc)
+        case QGRHanaDataTypes::Integer:
             ogrFieldType = columnDesc.isArray ? OFTIntegerList : OFTInteger;
             break;
-        case odbc::SQLDataTypes::BigInt:
+        case QGRHanaDataTypes::BigInt:
             ogrFieldType = columnDesc.isArray ? OFTInteger64List : OFTInteger64;
             break;
-        case odbc::SQLDataTypes::Double:
-        case odbc::SQLDataTypes::Real:
-        case odbc::SQLDataTypes::Float:
+        case QGRHanaDataTypes::Double:
+        case QGRHanaDataTypes::Real:
+        case QGRHanaDataTypes::Float:
             ogrFieldType = columnDesc.isArray ? OFTRealList : OFTReal;
-            if (columnDesc.type != odbc::SQLDataTypes::Double)
+            if (columnDesc.type != QGRHanaDataTypes::Double)
                 ogrFieldSubType = OGRFieldSubType::OFSTFloat32;
             break;
-        case odbc::SQLDataTypes::Decimal:
-        case odbc::SQLDataTypes::Numeric:
+        case QGRHanaDataTypes::Decimal:
+        case QGRHanaDataTypes::Numeric:
             ogrFieldType = columnDesc.isArray ? OFTRealList : OFTReal;
             setFieldPrecision = true;
             break;
-        case odbc::SQLDataTypes::Char:
-        case odbc::SQLDataTypes::VarChar:
-        case odbc::SQLDataTypes::LongVarChar:
-        case odbc::SQLDataTypes::WChar:
-        case odbc::SQLDataTypes::WVarChar:
-        case odbc::SQLDataTypes::WLongVarChar:
+        case QGRHanaDataTypes::Char:
+        case QGRHanaDataTypes::VarChar:
+        case QGRHanaDataTypes::LongVarChar:
+        case QGRHanaDataTypes::WChar:
+        case QGRHanaDataTypes::WVarChar:
+        case QGRHanaDataTypes::WLongVarChar:
             // Note: OFTWideString is deprecated
             ogrFieldType = columnDesc.isArray ? OFTStringList : OFTString;
             setFieldSize = true;
             break;
-        case odbc::SQLDataTypes::Date:
-        case odbc::SQLDataTypes::TypeDate:
+        case QGRHanaDataTypes::Date:
+        case QGRHanaDataTypes::TypeDate:
             ogrFieldType = OFTDate;
             break;
-        case odbc::SQLDataTypes::Time:
-        case odbc::SQLDataTypes::TypeTime:
+        case QGRHanaDataTypes::Time:
+        case QGRHanaDataTypes::TypeTime:
             ogrFieldType = OFTTime;
             break;
-        case odbc::SQLDataTypes::Timestamp:
-        case odbc::SQLDataTypes::TypeTimestamp:
+        case QGRHanaDataTypes::Timestamp:
+        case QGRHanaDataTypes::TypeTimestamp:
             ogrFieldType = OFTDateTime;
             break;
-        case odbc::SQLDataTypes::Binary:
-        case odbc::SQLDataTypes::VarBinary:
-        case odbc::SQLDataTypes::LongVarBinary:
+        case QGRHanaDataTypes::Binary:
+        case QGRHanaDataTypes::VarBinary:
+        case QGRHanaDataTypes::LongVarBinary:
             ogrFieldType = OFTBinary;
             setFieldSize = true;
             break;
+        case QGRHanaDataTypes::RealVector:
+            ogrFieldType = OFTBinary;
```

Comment:
Why not mapping this into as OGRRealList ? This would be much more user friendly

Reply:
Good point. I completely forgot about OGRRealList type, will consider your suggestion in the next commit.

Reply:
> By converting a double value back to a float may produce a value different from an original one.

no, as double has more precision than float, this will round to the original float value when converted back to float as the extra bits of the mantissa are dropped.   So float -> double -> float is lossless (but not double -> float -> double obviously)

> Hence, we would prefer to stick to the binary representation for the REAL_VECTOR type.

I'm fine with that too if that's your final choice

Reply:
You are right @rouault, I've confused myself.
So, the binary representation is our final decision.

At the moment, we don't see any urgent need in treating REAL_VECTOR type as OFTRealList. I guess that not all data source providers properly support OFTRealList.
Once there is a need in supporting OGRRealList or OFTString for REAL_VECTOR type, we can always implement it later and give a control via data source open options.
For example:
```
VECTOR_DATA_TYPE: Controls how REAL_VECTOR data type is represented. Possible values are: OFTBinary, OFTRealList or OFTString.
```
or something like that.

Thank you and kind regards